Disregard all about:<something> uris in the navigation policy.

pvoglhofer opened the pull request:
We already discovered issues with about:blank requests, as these are not
regular navigation requests. These peculiarities extend to other about:
requests as well, so they should be disregarded in the decide-policy
handling too.

I'm not entirely sure this will help #80 , but it is a potential source for hanging during the 'open' call (running iframe_open.t without the changes to WebKit2.pm produces a hang)

pvoglhofer committed:
Disregard all about:<something> uris in the navigation policy.

We already discovered issues with about:blank requests, as these are not
regular navigation requests. These peculiarities extend to other about:
requests as well, so they should be disregarded in the decide-policy
handling too.

niner commented on lib/WWW/WebKit2.pm:
```diff
@@ -361,8 +361,7 @@ sub init_webkit {
 
-            unless ($action_uri eq 'about:blank') {
-
+            unless ($action_uri =~ m/about:\w+$/) {
```
This would also find `http://test.example/about:me`
`m/\A about:/x` seems more appropriate

pvoglhofer replied:
True. Has been changed.

niner commented:
What happens if there is an active navigation request already and we navigate to about:blank? We can't set the active navigation request to about:blank but should we at least remove the existing (and obsolete) one?

Commit:
Extra test code, within iframe_open.

Unset active_navigation_action if it is set and we are trying to
navigate to an about* page and if a failed-with-tls-errors happens
within the resource-load-started signal.

jscarty commented:
> What happens if there is an active navigation request already and we navigate to about:blank? We can't set the active navigation request to about:blank but should we at least remove the existing (and obsolete) one?


Depends on how that happens. If it is set and WWW::Webkit2 activates something it will most likely get stuck inside process page load, nothing is loading so is_loading and Gtk3::events_pending will return false while active_navigation_action will always return true.

I can't think of a situation where it will be set and then navigation to an about:* page happens. I assume it should be unset when resouce-load-signal is activated.
